src/adapter/etl-adapter-google-sheet/src/Flow/ETL/Adapter/GoogleSheet/RowsNormalizer/EntryNormalizer.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\GoogleSheet\RowsNormalizer;
+
+use function Flow\ETL\DSL\{date_interval_to_microseconds, type_json};
+use Flow\ETL\PHP\Type\Caster;
+use Flow\ETL\Row\Entry;
+
+final readonly class EntryNormalizer
+{
+    public function __construct(
+        private Caster $caster,
+        private string $dateTimeFormat = \DateTimeInterface::ATOM,
+        private string $dateFormat = 'Y-m-d',
+    ) {
+    }
+
+    /**
+     * @param Entry<mixed, mixed> $entry
+     */
+    public function normalize(Entry $entry) : string|float|int|bool|null
+    {
+        return match ($entry::class) {
+            Entry\UuidEntry::class,
+            Entry\XMLElementEntry::class,
+            Entry\XMLEntry::class => $entry->toString(),
+            Entry\DateTimeEntry::class => $entry->value()?->format($this->dateTimeFormat),
+            Entry\DateEntry::class => $entry->value()?->format($this->dateFormat),
+            Entry\TimeEntry::class => $entry->value() ? date_interval_to_microseconds($entry->value()) : null,
```

Comment:
I believe excel has a decent format for intervals that can be rendered like `00:00`